src/snd_card_plugin.c

```diff
@@ -125,19 +111,23 @@ struct snd_node *snd_utils_get_dev_node(unsigned int card,
     if (rc < 0)
         goto err_resolve_symbols;
 
-    node->card_node = node->open_card(card);
+    node->card_node = node->ops->open_card(card);
     if (!node->card_node)
         goto err_resolve_symbols;
 
-    node->dev_node = node->get_node(node->card_node,
```

Comment:
The purpose of get_node() was to use it extend it for compress plugin. Making it to get_pcm/get_mixer limits the scope of snd card utils.

Reply:
Ah, I see. The type-safe way is advantageous because for future functions, the parameter list to the `*_open` function can be declared specifically for the interface being constructed. The type-safe way can be used for a potential tinycompress function later on by adding a `void* reserved` that can later be rewritten.

For example:
```c
struct snd_node_ops {
  /* ... */
  /* Reserved pointers for future use.
   * Must be set to NULL. */
  void* reserved[4];
};
```
Changed later on to:
```c
struct snd_node_ops {
  /* ... */
  void* (*get_compress)(...);
  void* reserved[3];
};
```